<?xml version="1.0"?>
<!--
Copyright (c) 2016, Picodex Robotics
All rights reserved.

Redistribution and use in source and binary forms, with or without
modification, are permitted provided that the following conditions are met:
    * Redistributions of source code must retain the above copyright
      notice, this list of conditions and the following disclaimer.
    * Redistributions in binary form must reproduce the above copyright
      notice, this list of conditions and the following disclaimer in the
      documentation and/or other materials provided with the distribution.
    * Neither the name of Clearpath Robotics nor the
      names of its contributors may be used to endorse or promote products
      derived from this software without specific prior written permission.

THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDERS BE LIABLE FOR ANY
DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
(INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
-->
<launch>
	
  
     <arg name="port_driver" default="$(optenv ULTRON_PORT_BASE /dev/ttyUSB0)" />
  <!--   <arg name="port_imu" default="$(optenv ULTRON_PORT_IMU /dev/ttyUSB1)" />-->

     <!--params -->
   <!--  <rosparam file="$(find ultron_kernel)/config/rc_hobbyking.yaml" />-->

       <!-- Petrorov arduino driver -->
       <!-- ATTENZIONE broadcast_tf va conflitto con ekf_localization ??
        leggere enable_odom_tf dalle properties -->	
  <!--   <node pkg="ultron_kernel" type="mpu6050_serial_to_imu_node" name="mpu6050_serial_to_imu_node" required="true">
        <param name="port" value="$(arg port_imu)"/>
        <param name="tf_parent_frame_id" value="imu_link"/>
        <param name="tf_frame_id" value="base_link"/>
        <param name="frame_id" value="imu_link"/>
        <param name="time_offset_in_seconds" value="0"/>
	<param name="broadcast_tf" value="false"/>
	<param name="broadcast_odom" value="false"/>
	<param name="linear_acceleration_stddev" value="0"/>
	<param name="angular_velocity_stddev" value="0"/>
	<param name="orientation_stddev" value="0"/>
	<remap from="joy" to="joy_teleop/joy" />
    </node>
-->
     <!-- rate are in HZ -->
     <!-- baud: 115200,57600 -->
   <!-- <node pkg="ultron_kernel" type="myserial_node.py" name="petrorov_node">-->
    <node pkg="rosserial_python" type="serial_node.py" name="petrorov_node">
        <rosparam subst_value="true">
            port: $(arg port_driver)
            baud: 115200

            max_acceleration: 3.0
            max_speed: 1.0
            wheel_diameter: 0.15

            polling_timeout: 2.0
            
            motor_controller_rate : 60.0
            odom_state_rate : 10
            diagnostic_rate: 1.0
            range_sensor_rate: 10.0
            
            Kp : 10
            Kd : 0
            Ki : 0
            Ko : 0
        </rosparam>
    </node>


</launch>
